test/screens/alerts/cache/filter_test.exs

```diff
@@ -66,4 +67,120 @@ defmodule Screens.Alerts.Cache.FilterTest do
+        alert("stop and route", ie(stop: "stop-and-route-stop", route: "stop-and-route-route")),
+        alert("route only 2", ie(route: "route-only-2", route_type: 1)),
+        alert("route type only", ie(route_type: 0)),
+        alert(
+          "route in one direction",
+          ie(route: "route-in-direction", direction_id: 0, route_type: 3)
+        ),
+        alert(
+          "route in the other direction",
+          ie(route: "route-in-direction", direction_id: 1, route_type: 3)
+        ),
+        alert("other activities 1", ie(route: "route-only", activities: ~w[USING_WHEELCHAIR])),
+        alert("other activities 2", ie(route: "route-only-3", activities: ~w[USING_WHEELCHAIR]))
+      ]
+
+      %{alerts: alerts}
+    end
+
+    test "filters for stop, including alerts impacting routes serving the stop", %{alerts: alerts} do
+      stub(Route.Mock, :serving_stop, fn _ -> {:ok, [%Route{id: "serving-stop"}]} end)
+
+      assert [%Alert{id: "stop only"}, %Alert{id: "route serving stop"}] =
+               Filter.filter_by(alerts, %{stops: ["stop-only"]})
+    end
+
+    test "filters for route, including alerts impacting the entire route type", %{alerts: alerts} do
+      stub(Route.Mock, :by_id, fn
+        "route-only" -> {:ok, %Route{id: "route-only", type: :light_rail}}
+      end)
+
+      assert [%Alert{id: "route only"}, %Alert{id: "route type only"}] =
+               Filter.filter_by(alerts, %{routes: ["route-only"]})
+    end
+
+    test "filters for stop and route combined", %{alerts: alerts} do
+      stub(Route.Mock, :by_id, fn
+        "stop-and-route-route" -> {:ok, %Route{id: "stop-and-route-route", type: :subway}}
+        "route-only-2" -> {:ok, %Route{id: "route-only", type: :subway}}
+      end)
+
+      stub(Route.Mock, :serving_stop, fn "stop-and-route-stop" ->
+        {:ok, [%Route{id: "stop-and-route-route"}]}
+      end)
+
+      assert [%Alert{id: "stop and route"}, %Alert{id: "route only 2"}] =
+               Filter.filter_by(alerts, %{
+                 routes: ["stop-and-route-route", "route-only-2"],
+                 stops: ["stop-and-route-stop"]
+               })
+    end
+
+    test "filters for route_type", %{alerts: alerts} do
+      assert [%Alert{id: "route only"}, %Alert{id: "route type only"}] =
+               Filter.filter_by(alerts, %{route_types: [0]})
+
+      assert [%Alert{id: "route only 2"}] = Filter.filter_by(alerts, %{route_types: [1]})
+
+      assert [%Alert{id: "route only"}, %Alert{id: "route only 2"}, %Alert{id: "route type only"}] =
+               Filter.filter_by(alerts, %{route_types: [0, 1]})
+    end
+
+    test "filters for a specific direction_id", %{alerts: alerts} do
+      stub(Route.Mock, :by_id, fn
+        "route-in-direction" -> {:ok, %Route{id: "route-in-direction", type: :bus}}
+      end)
+
+      assert [%Alert{id: "route in one direction"}] =
+               Filter.filter_by(alerts, %{routes: ["route-in-direction"], direction_id: 0})
```

Comment:
Shouldn't this also include all alerts with a "non-directional" informed entity? Since its absence implies the alert affects both directions of service. What I'd expect the direction filter to do is basically reject any alerts that _only_ affect the opposite direction. (Though I don't know off-hand if this is how the API does it...)

Reply:
You're correct. That is how this code and the API handle that. I've updated the test case to reflect that.

Reply:
Ah, thanks for the update, I was a little confused about how the test data intersected with the specific test case.